jvb/src/main/resources/reference.conf

```diff
@@ -177,37 +177,36 @@ videobridge {
   stats {
     # Whether periodic gathering of statistics is enabled or not. When enabled they are accessible through the REST
     # API (at `/colibri/stats`), and are available to other modules (e.g. to be pushed to callstats or in a MUC).
-    #
     enabled = false
 
     # The interval at which stats are gathered.
     interval = 5 seconds
-  }
-  # Configuration related to pushing statistics to callstats.io.
-  callstats {
-    # An integer application ID (use 0 to disable pushing stats to callstats).
-    app-id = 0
 
-    # The shared secred to authentication with callstats.io.
-    //app-secret = "s3cret"
+    # Configuration related to pushing statistics to callstats.io.
+    callstats {
+      # An integer application ID (use 0 to disable pushing stats to callstats).
+      app-id = 0
 
-    # ID of the key that was used to generate token.
-    //key-id = "abcd"
+      # The shared secred to authentication with callstats.io.
+      //app-secret = "s3cret"
 
-    # The path to private key file.
-    //key-path = "/etc/jitsi/videobridge/ecpriv.jwk"
+      # ID of the key that was used to generate token.
+      //key-id = "abcd"
 
-    # The ID of the server instance to be used when reporting to callstats.
-    bridge-id = "jitsi"
+      # The path to private key file.
+      //key-path = "/etc/jitsi/videobridge/ecpriv.jwk"
 
-    # TODO: document
-    //conference-id-prefix = "abcd"
+      # The ID of the server instance to be used when reporting to callstats.
+      bridge-id = "jitsi"
 
-    # The interval at which statististics will be published to callstats. This affects both per-conference and global
-    # statistics.
-    # Note that this value will be overriden if a "callstatsio" transport is defined in the "stats" sections above.
-    interval = ${videobridge.stats.interval}
+      # TODO: document
+      //conference-id-prefix = "abcd"
 
+      # The interval at which statististics will be published to callstats. This affects both per-conference and global
+      # statistics.
+      # Note that this value will be overriden if a "callstatsio" transport is defined in the parent "stats" section.
```

Comment:
Will the transport still be scoped under a `transports` block? I.e.:
```
videobridge {
    stats {
        transports {
            callstatsio {
            }
        }
    }
}
```

Reply:
No, no, no. The "transports" are gone. We just read them in the old format (`transports: [ ... ]`) for backward compat.

To configure callstats use the `callstats` section. To configure the interval for MUC presence updates use `videobridge.apis.xmpp-client.presence-interval`. The fact that under the hood these are implemented as "transports" in a stats manager is irrelevant and isn't exposed in the config file anymore.

Reply:
OK, and I see we no longer explicitly enable the muc stats transport, the code just always initialize it?  And uses the xmpp connection's presence interval for the stats interval instead of allowing it to be configured?  What if someone isn't using the XMPP connection?

Reply:
> OK, and I see we no longer explicitly enable the muc stats transport, the code just always initialize it?

Yes. The XMPP connection without presence updates is not very useful.

> And uses the xmpp connection's presence interval for the stats interval instead of allowing it to be configured?

No. The "stats interval" is the interval at which stats are gathers. It's controlled through `stats.interval` (I haven't changed that). 

There used to be per-transport intervals (meaning how often stats are pushed to that transport). These are now configurable through the module that uses them (`videobridge.apis.xmpp-client.presence-interval` and `videobridge.stats.callstats.interval`), and for backwards-compat they default to the stats interval.

>  What if someone isn't using the XMPP connection?

I think if the XMPP connection is not used, then it shouldn't subscribe to stats, though stats should still be generated as configured in `videobridge.stats` to e.g. expose via HTTP. I now see that `XmppConnection` is created and started unconditionally, so that won't work. I'll make a change.

Reply:
Actually, that won't work, since MUC clients can be added dynamically over HTTP. The answer is that the current code does not allow you to not have an XMPP connection. I think this should change, but not in this PR.

Reply:
> There used to be per-transport intervals (meaning how often stats are pushed to that transport). These are now configurable through the module that uses them (videobridge.apis.xmpp-client.presence-interval and videobridge.stats.callstats.interval), and for backwards-compat they default to the stats interval.

Ok, this is what I meant.  I do think it's a little implicit that the xmpp stats publish interval is configured via the presence interval, but probably not terrible.

> Actually, that won't work, since MUC clients can be added dynamically over HTTP. The answer is that the current code does not allow you to not have an XMPP connection. I think this should change, but not in this PR.

Ok

Reply:
>  I do think it's a little implicit that the xmpp stats publish interval is configured via the presence interval

XMPP stats publish *is* the presence update

Reply:
Yes but presence could be sent for other reasons and not contain any stats as well.  And it's not inherent that presence contains stats, we just happen to implement it that way, no?

Reply:
At the XMPP level, I mean.  I'm not saying it's wrong, it's just an implicit implementation detail that I don't think would be obvious.  If I wanted to know how often stats were sent, I wouldn't think to look for the presence update interval.

Reply:
The other reason I'm thinking this is: imagine we add another stats service here (RTCStats, maybe): we'd configure that interval within the RTCStats configuration, so it seems like these intervals make sense to be configured alongside the stats service, not whatever transport they may be using.

I agree this is potentially awkward for the 'presence interval' config value: as it would effectively be controlled by 2 config values (one in the client and one here in stats).  I'm not sure what the best course of action would be there.

Reply:
> At the XMPP level, I mean. I'm not saying it's wrong, it's just an implicit implementation detail that I don't think would be obvious. If I wanted to know how often stats were sent, I wouldn't think to look for the presence update interval.

Would it be more clear to name the property `stats-interval` instead of `presence-interval`?

Reply:
> we'd configure that interval within the RTCStats configuration, so it seems like these intervals make sense to be configured alongside the stats service, not whatever transport they may be using.

I don't follow. Why does it make sense to configure the interval alongside the stats service? How does it make sense for the stats manager, which now just generates stats and notifies entities that are subscribed to it, to know that a subscriber is RTCStats and therefore it should use a particular interval for it? That's exactly what I'm trying to get away from.

Reply:
> I don't follow. Why does it make sense to configure the interval alongside the stats service? How does it make sense for the stats manager, which now just generates stats and notifies entities that are subscribed to it, to know that a subscriber is RTCStats and therefore it should use a particular interval for it? That's exactly what I'm trying to get away from.

Sorry, I'm not suggesting that.  I'm suggesting the stats publish interval to some stats service be configured alongside the rest of the configuration for that stats service.

Reply:
That's exactly what this PR does

Reply:
Sorry, I do think I forgot that the Callstats interval is alongside the Callstats config and that the muc stats don't have any explicit config, but I was referring to the fact that the MUC stats interval is configured via `videobridge.apis.xmpp-client.presence-interval`, as opposed to something like: `videobridge.stats.muc.interval`.